x/examples/fetch-psiphon/main.go

```diff
@@ -80,7 +80,9 @@ func main() {
 
 	// Start the Psiphon dialer.
 	dialer := psiphon.GetSingletonDialer()
-	if err := dialer.Start(context.Background(), config); err != nil {
+	ctx, cancel := context.WithTimeout(context.Background(), 30 * time.Second)
```

Comment:
This is not a dial timeout. It's a start timeout.
It's not how Psiphon is intended to work. It's supposed to keep running in the background.

Though, it should fail quickly if the config is invalid. Can you file a bug on their repository?

Reply:
done.

From trawling their code I don't think it's a bug. The 5m timeout is set deliberately [here](https://github.com/Psiphon-Labs/psiphon-tunnel-core/blob/ca5fc873ba9cece39a8b7290e054a95cdccc912b/psiphon/common/parameters/parameters.go#L547). I guess they set it so high to allow for really adverse network condition. But our fetch tool doesn't need to be so accommodating.

I didn't spend too long looking at the [psiphon design](https://github.com/Psiphon-Inc/psiphon/blob/master/files/DESIGN.pdf), but I'm assuming they simply don't respond to made up IDs because they want to avoid an adversary being able to enumerate all the IDs.

You've definitely spend more time looking at the psiphon design than I have, do you know something that would contradict that?